### ISSUE TITLE
Debug branch: assertion violation lar_solver.cpp:828

### ISSUE THREAD

**Issue by rainoftime** [Z3Prover/z3#3405]
Hi, for the following opt instances, z3 debug branch (Commit 9b58b7727466) throws out an assertion 
[lar828.txt](https://github.com/Z3Prover/z3/files/4352801/lar828.txt)

~~~~
File: ../src/math/lp/lar_solver.cpp
Line: 828
m_mpq_lar_core_solver.m_r_solver.reduced_costs_are_correct_tableau()
~~~~
Since the crash is at  lar_solver.cpp, I report as "debug branch".

In the master branch (e075f381527), z3 also crashes at the same file 
~~~~
File: ../src/math/lp/lar_solver.cpp
Line: 429
jset.m_index.size()==0
~~~~

**Comment by muchang** [comment]
Here is a smaller test case for triggering lar_solver.cpp assertion violation on debug branch:
```
$ cat small.smt2 
(declare-fun a () Real)    
(assert (<= (div 0 0) 1)) 
(minimize a) 
(check-sat)    
$ z3debug/build/z3 small.smt2
ASSERTION VIOLATION
File: ../src/math/lp/lar_solver.cpp
Line: 828
m_mpq_lar_core_solver.m_r_solver.reduced_costs_are_correct_tableau()
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
A
```

**Comment by levnach** [comment]
Fixed in  1ffa5df9345ccd44a6f8947e1cfc434201575833
